resource/subfunctions/resource_class.md

```diff
@@ -0,0 +1,323 @@
+     "cdtinfo:valid_other_characters": ["ALPHA" , "NUMERIC"],
+     "cdtinfo:max_length": 246,
+     "cdtinfo:max_length_entityx": 246,
+     "cdtinfo:key_qualifiers": 0,
+     "cdtinfo:profiles_allowed": "YES",
+     "cdtinfo:posit_number": 200,
+     "cdtinfo:default_racroute_return_code": 8,
+     "cdtinfo:default_universal_access": "NONE",
+     "cdtinfo:raclist_allowed": "ALLOWED",
+}
+resource_admin.add_resource_class("SHELCITY",traits=class_traits)
+{'securityResult': {'resource': {'name': 'SHELCITY', 'class': 'CDT', 'operation': 'set', 'requestId': 'ResourceRequest', 'commands': [{'safReturnCode': 0, 'returnCode': 0, 'reasonCode': 0, 'image': 'RDEFINE CDT             (SHELCITY) ', 'messages': ['ICH10006I RACLISTED PROFILES FOR CDT WILL NOT REFLECT THE ADDITION(S) UNTIL A SETROPTS REFRESH IS ISSUED.', 'IRR52205I Warning: CDTINFO segment is required for the CDT class.']}, {'safReturnCode': 0, 'returnCode': 0, 'reasonCode': 0, 'image': 'RALTER  CDT             (SHELCITY)  CDTINFO  (CASE        (UPPER) FIRST       (ALPHA) OTHER       (ALPHA NUMERIC) MAXLENX     (246) MAXLENGTH   (246) KEYQUALIFIER(0) PROFILESALLO(YES) POSIT       (200) DEFAULTRC   (8) DEFAULTUACC (NONE) RACLIST     (ALLOWED))', 'messages': ['IRR52199I Warning: Class name SHELCITY does not contain a national character nor a number.', 'ICH11009I RACLISTED PROFILES FOR CDT WILL NOT REFLECT THE UPDATE(S) UNTIL A SETROPTS REFRESH IS ISSUED.']}]}, 'returnCode': 0, 'reasonCode': 0}}
+```
+
+###### Security Result Dictionary as JSON
+```json
+{
+  "securityResult":{
+    "resource":{
+      "name":"SHELCITY",
+      "class":"CDT",
+      "operation":"set",
+      "requestId":"ResourceRequest",
+      "commands":[
+        {
+          "safReturnCode":0,
+          "returnCode":0,
+          "reasonCode":0,
+          "image":"RDEFINE CDT             (SHELCITY) ",
+          "messages":[
+            "ICH10006I RACLISTED PROFILES FOR CDT WILL NOT REFLECT THE ADDITION(S) UNTIL A SETROPTS REFRESH IS ISSUED.",
+            "IRR52205I Warning: CDTINFO segment is required for the CDT class."
+          ]
+        },
+        {
+          "safReturnCode":0,
+          "returnCode":0,
+          "reasonCode":0,
+          "image":"RALTER  CDT             (SHELCITY)  CDTINFO  (CASE        (UPPER) FIRST       (ALPHA) OTHER       (ALPHA NUMERIC) MAXLENX     (246) MAXLENGTH   (246) KEYQUALIFIER(0) PROFILESALLO(YES) POSIT       (200) DEFAULTRC   (8) DEFAULTUACC (NONE) RACLIST     (ALLOWED))",
+          "messages":[
+            "IRR52199I Warning: Class name SHELCITY does not contain a national character nor a number.",
+            "ICH11009I RACLISTED PROFILES FOR CDT WILL NOT REFLECT THE UPDATE(S) UNTIL A SETROPTS REFRESH IS ISSUED."
+          ]
+        }
+      ]
+    },
+    "returnCode":0,
+    "reasonCode":0
+  }
+}
+```
+
+## `ResourceAdmin.alter_resource_class()`
+
+```python
+  def alter_resource_class( self, class_name: str, traits: dict = {} ) -> Union[dict, bytes]:
+```

```

Comment:
```python
def alter_resource_class(self, class_name: str, traits: dict = {}) -> Union[dict, bytes]:
```

Reply:
Remove leading spaces.

Reply:
Leading spaces and unnecessary trailing spaces and indents removed across all function definitions.